crates/matrix-sdk/src/sliding_sync/list/builder.rs

```diff
@@ -143,20 +144,23 @@ impl SlidingSyncListBuilder {
     }
 
     /// Set the ranges to fetch.
-    pub fn ranges<U: Into<UInt>>(mut self, range: Vec<(U, U)>) -> Self {
-        self.ranges = range.into_iter().map(|(a, b)| (a.into(), b.into())).collect();
+    pub fn ranges<U: Into<u32> + Copy>(mut self, range: Vec<RangeInclusive<U>>) -> Self {
```

Comment:
Ah it's a good idea, and it would avoid a few dumb allocations here and there (it might also avoid some obvious type hints that rustc couldn't infer based on the generic parameters).

Reply:
> I'm pretty sure the only reason we used impl Into<UInt> was to avoid having to write uint!().

Indeed, that was the reason.